base/steer/FairTSBufferFunctional.h:35:35: struct std::binary_function’ is deprecated [-Wdeprecated-declarations]

```cpp
[116/509] Building CXX object base/CMakeFiles/Base.dir/steer/FairTSBufferFunctional.cxx.o
In file included from /home/dklein/projects/FairRoot2/base/steer/FairTSBufferFunctional.cxx:8:
/home/dklein/projects/FairRoot2/base/steer/FairTSBufferFunctional.h:35:35: warning: ‘template<class _Arg1, class _Arg2, class _Result> struct std::binary_function’ is deprecated [-Wdeprecated-declarations]
   35 | class BinaryFunctor : public std::binary_function<FairTimeStamp*, double, bool>
      |                                   ^~~~~~~~~~~~~~~
In file included from /usr/include/c++/12/string:48,
                 from /home/dklein/fairsoft/apr22/include/root/TSchemaHelper.h:17,
                 from /home/dklein/fairsoft/apr22/include/root/TGenericClassInfo.h:21,
                 from /home/dklein/fairsoft/apr22/include/root/Rtypes.h:191,
                 from /home/dklein/projects/FairRoot2/base/event/FairLink.h:26,
                 from /home/dklein/projects/FairRoot2/base/event/FairMultiLinkedData_Interface.h:11,
                 from /home/dklein/projects/FairRoot2/base/event/FairTimeStamp.h:11,
                 from /home/dklein/projects/FairRoot2/base/steer/FairTSBufferFunctional.h:11:
/usr/include/c++/12/bits/stl_function.h:131:12: note: declared here
  131 |     struct binary_function
      |            ^~~~~~~~~~~~~~~
```

https://en.cppreference.com/w/cpp/utility/functional/binary_function - technically already removed in C++17

--- a/base/steer/FairTSBufferFunctional.h
+++ b/base/steer/FairTSBufferFunctional.h
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -32,9 +32,13 @@ class TTree;
  * The method TimeOut is used to break the processing if for example always the same data is requested.
  */
 
-class BinaryFunctor : public std::binary_function<FairTimeStamp*, double, bool>
+class BinaryFunctor
 {
   public:
+    using first_argument_type = FairTimeStamp*;
+    using second_argument_type = double;
+    using result_type = bool;
+
     virtual bool operator()(FairTimeStamp* a, double b) { return Call(a, b); };
     virtual bool Call(FairTimeStamp* a, double b) = 0;
     virtual bool TimeOut() { return false; }

